src/Sylius/Component/Core/Dashboard/SalesSummary.php

```diff
@@ -19,39 +19,41 @@
 final class SalesSummary implements SalesSummaryInterface
 {
     /** @psalm-var array<string, string> */
-    private $monthsSalesMap = [];
+    private $intervalsSalesMap = [];
 
     public function __construct(
         \DateTimeInterface $startDate,
         \DateTimeInterface $endDate,
-        array $salesData
+        string $interval,
+        array $salesData,
+        string $dateFormat
     ) {
-        $period = new \DatePeriod($startDate, \DateInterval::createFromDateString('1 month'), $endDate);
+        $dateIntervals = new \DatePeriod($startDate, \DateInterval::createFromDateString(sprintf('1 %s', $interval)), $endDate);
```

Comment:
Do we restrict what kind of intervals do we support? Maybe we should have some kind of enum for them?